lib/helpers.js

```diff
@@ -98,20 +98,18 @@ exports.validateOptionalName = function (name) {
  * Check if relation has a different name from properties
  * @param {Object} modelDefinition The model which has the relation
  * @param {String} name The user input
- * @param {Function(String|Boolean)} Callback to receive the check result.
+ * @returns {String|Boolean} Return the check result.
```

Comment:
My understanding is that returning a promise (instead of calling a callback) is the whole point of this patch/change, right?

In the code, you do `return getAsync().then(..)`, which returns a promise.